romancal/pipeline/exposure_pipeline.py

```diff
@@ -71,66 +75,100 @@ def process(self, input):
             input_filename = None
 
         # open the input file
-        input = rdd.open(input)
-
-        log.debug("Exposure Processing a WFI exposure")
-
-        self.dq_init.suffix = "dq_init"
-        result = self.dq_init(input)
-        if input_filename:
-            result.meta.filename = input_filename
-        result = self.saturation(result)
-
-        # Test for fully saturated data
-        if is_fully_saturated(result):
-            log.info("All pixels are saturated. Returning a zeroed-out image.")
-
-            # Return zeroed-out image file (stopping pipeline)
-            return self.create_fully_saturated_zeroed_image(result)
-
-        result = self.linearity(result)
-        result = self.dark_current(result)
-        result = self.jump(result)
-        result = self.rampfit(result)
-
-        # Test for fully saturated data
-        if "groupdq" in result.keys():
+        file_type = filetype.check(input)
+        asn = None
+        if file_type == "asdf":
+            try:
+                input = rdm.open(input)
+            except TypeError:
+                log.debug("Error opening file:")
```

Comment:
What should happen in the case of one asdf file which cannot be opened? Should it return?

Reply:
I guess the question is, if any of the inputs in the association fail for any reason will the end user still want the results? 
My choice would be to stop and let the user know. They can fix the error and rerun the pipeline. That way they make the decision if the results are valid without all the inputs. Not very pythonic of me.